students/jerickson/Lesson3/mailroom.py

```diff
@@ -0,0 +1,154 @@
+        print(donor_string)
+    print(report_end)
+
+
+def donor_existance(donor_name):
+    """Returns a donor's record if it exists, else None"""
+    return_record = None  # TODO Remove this
+    for donor_record in donation_data:
+        if donor_name.lower() == donor_record[0].lower():
+            return_record = donor_record
+            break
+    else:
+        return_record = [donor_name, 0, 0]
+        donation_data.append(return_record)  # Create donor if not found
+    return return_record
+
+
+def new_donation(donor_name, amount):
+    """Adds a new donation to the donation record"""
+    donor_record = donor_existance(donor_name)
+    donor_record[1] += amount
+    donor_record[2] += 1
+    compose_email(donor_record, amount)
+
+
+def donor_list():
+    """Return a string of comma seperated donor names"""
+    all_names = []
+    for donor_record in donation_data:
+        all_names.append(donor_record[0])
+    format_string = " {}," * (len(all_names))
+    return format_string.format(*all_names)[:-1]  # Slice removes extra comma at end
+
+
+def compose_email(donor_record, amount):
+    """Print a thank-you email using the donor's historical information"""
+    time_s = "times" if donor_record[2] > 1 else "time"
+    email = f"Thank you {donor_record[0]} for your donation of ${amount:.2f}! You have donated {donor_record[2]} {time_s} for a total of ${donor_record[1]:.2f}."
+    print(email)
+
+
+def thank_you():
+    """Add new donation to the record and compose a thank you email for it"""
+    donor_name = None
+    while not donor_name:
+        donor_name = input("Who just made a donation? Full Name please. ->: ")
+        if donor_name == "list":
+            print("All Donors:" + donor_list())
+            donor_name = None
+    donor_amount = float(input("How much was the donation? ->: "))
+    new_donation(donor_name, donor_amount)
+
+
+if __name__ == "__main__":
+    print("\nBack to the grind in the mailroom.", end="\n\n")
+    are_you_mocking_me = int(input('Are you mocking me?? "0": no, "1": yes->: '))
+
+    if are_you_mocking_me:
+        """Mocks input() to allow for automated list of user-inputs to be run"""
+
+        def response_generator(seq):
+            for item in seq:
+                yield item
+
+        def input(prompt):  # Mocks input function
+            print(prompt, end="")
+            response = mocked_resp_gen.__next__()
+            print(response)
+            return response
+
+        mocked_responses = [
+            "spam",
+            "create a report",
+            "send a thank you",
+            "Bob Barker",
+            "1",
+            "send a thank you",
+            "list",
+            "King ArThur",
+            "400.2",
+            "create a report",
+            "quit",
+        ]
+        mocked_resp_gen = response_generator(mocked_responses)
+
+    quit_flag = False
+    while not quit_flag:
```

Comment:
you can simplify this a bit by using a `while True` with a `break` statement; avoids using a flag variable.

Reply:
Incorporated into Lesson04 mailroom, thank you.